utils/profileUtils.js

```diff
@@ -0,0 +1,18 @@
+import { getSessionStorage, setSessionStorage } from './sessionStorageUtils.js';
+
+const PROFILE_URL = '/services/profile';
+
+const loadProfileDetails = async () => {
+  let profileData = getSessionStorage('profile');
+  if (!profileData) {
+    try {
+      const response = await fetch(PROFILE_URL, { headers: { 'Content-Type': 'application/json' } });
+      profileData = await response.json();
+    } catch (e) {
+      console.log('error in fetching the user data', e);
+    }
+  }
+  setSessionStorage('profile', profileData?.details);
```

Comment:
You may want to set Session Storage with profile details ONLY when the data from profile API is available (and not when an error has occurred); so maybe it could be moved after L#10 (after `profileData = await response.json();`). 

Right now it will try to set `profile` key to `undefined` if the data from API is not available (unless this is what you want)